File: 品味Java的21種設計模式/dp_java/memento/example3/FlowAMementoCareTaker.java
package cn.javass.dp.memento.example3;


/**
 * 负责保存模拟运行流程A的对象的备忘录对象
 */
public class FlowAMementoCareTaker {
    /**
     * 记录被保存的备忘录对象
     */
    private FlowAMockMemento memento = null;
    /**
     * 保存备忘录对象
     * @param memento 被保存的备忘录对象
     */
    public void saveMemento(FlowAMockMemento memento) {
        this.memento = memento;
    }
    /**
     * 获取被保存的备忘录对象
     * @return 被保存的备忘录对象
     */
    public FlowAMockMemento retriveMemento() {
        return this.memento;
    }
}

File: 品味Java的21種設計模式/dp_cpp/memento/example3/FlowAMementoCareTaker.cpp
#include "FlowAMementoCareTaker.h"

namespace cn {
namespace javass {
namespace dp {
namespace memento {
namespace example3 {

void FlowAMementoCareTaker::saveMemento(FlowAMockMemento *memento) {
    this->memento = memento;
}

FlowAMockMemento *FlowAMementoCareTaker::retriveMemento() {
    return this->memento;
}
}
}
}
}
}
